src/dygraph.js

```diff
@@ -2026,14 +2026,23 @@ Dygraph.prototype.predraw_ = function() {
 
   this.cascadeEvents_('predraw');
 
+  var seriesName = this.getLabels();
+
   // Convert the raw data (a 2D array) into the internal format and compute
   // rolling averages.
   this.rolledSeries_ = [null];  // x-axis is the first series and it's special
   for (var i = 1; i < this.numColumns(); i++) {
     // var logScale = this.attr_('logscale', i); // TODO(klausw): this looks wrong // konigsberg thinks so too.
     var series = this.dataHandler_.extractSeries(this.rawData_, i, this.attributes_);
-    if (this.rollPeriod_ > 1) {
-      series = this.dataHandler_.rollingAverage(series, this.rollPeriod_, this.attributes_);
+	var seriesRollPeriod;
+	
+	if (this.getOption('rollPeriod', seriesName[i])) { 
+		seriesRollPeriod = this.getOption('rollPeriod', seriesName[i]);
```

Comment:
It should suffice to say:

```js
const rollPeriod = this.getOption('rollPeriod', seriesName[i]);
```

this will get the per-series value or fall back to the global one.